manifests/init.pp

```diff
@@ -128,31 +128,48 @@
         user     => root,
         provider => shell,
         creates  => $repo_location,
-    } ->
+    }
+    if ($package_type == 'deb') {
+        file { '/etc/apt/preferences.d/pin-gitlab-runner.pref':
+            ensure  => 'present',
+            mode    => '0644',
+            content => 'Explanation: Prefer GitLab provided packages over the Debian native ones
+Package: gitlab-runner
+Pin: origin packages.gitlab.com
+Pin-Priority: 1001',
+            require => Exec['Add Repository']
+        }
+    }
+
     # Install the package after the repo has been added.
-    package { 'gitlab-ci-multi-runner':
-        ensure => $theVersion,
-    } ->
+    package { 'gitlab-runner':
+        ensure  => $theVersion,
+        require => Exec['Add Repository']
+    }
     exec { 'Uninstall Misconfigured Service':
         command  => "service ${service} stop; ${service} uninstall",
         user     => root,
         provider => shell,
         unless   => "grep '${toml_file}' ${service_file}",
-    } ->
+        require  => Package['gitlab-runner']
+    }
     exec { 'Ensure Service':
         command  => "${service} install --user ${user} --config ${toml_file} --working-directory ${home_path}",
         user     => root,
         provider => shell,
         creates  => $service_file,
-    } ->
+        require  => Exec['Uninstall Misconfigured Service']
+    }
     file { 'Ensure .gitlab-runner directory is owned by correct user':
         path    => $toml_path,
         owner   => $user,
         recurse => true,
-    } ->
+        require => Exec['Ensure Service']
+    }
     # Ensure that the service is running at all times.
     service { $service:
```

Comment:
What about the update post install script for Redhat/CentOS/Fedora described in [GitLab Runner Install Documentation](https://docs.gitlab.com/runner/install/linux-repository.html#upgrading-to-gitlab-runner-10)?